File: apps/BuddyPlay/android/core/src/main/java/com/americangroupllc/buddyplay/core/models/Transport.kt
package com.americangroupllc.buddyplay.core.models

import kotlinx.serialization.Serializable

/**
 * The transport an active game session is using. Surfaced in the lobby so
 * the user knows whether they're on Wi-Fi, Hotspot, or BLE.
 */
@Serializable
enum class Transport {
    WIFI, HOTSPOT, BLE;

    val displayName: String
        get() = when (this) {
            WIFI    -> "Local Wi-Fi"
            HOTSPOT -> "Mobile Hotspot"
            BLE     -> "Bluetooth"
        }

    val supportsRealtime: Boolean
        get() = this != BLE
}

File: apps/BuddyPlay/shared/BuddyCore/Sources/BuddyCore/Models/Transport.swift
import Foundation

/// The transport an active game session is using. Surfaced in the lobby so
/// the user knows whether they're on Wi-Fi, Hotspot, or BLE.
public enum Transport: String, Codable, Sendable {
    case wifi
    case hotspot
    case ble

    public var displayName: String {
        switch self {
        case .wifi:    return "Local Wi-Fi"
        case .hotspot: return "Mobile Hotspot"
        case .ble:     return "Bluetooth"
        }
    }

    /// Quality hint for the lobby. Real-time games (Racer) refuse `.ble`.
    public var supportsRealtime: Bool {
        switch self {
        case .wifi, .hotspot: return true
        case .ble:            return false
        }
    }
}
